packages/select/src/Select/components/Root/Root.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react"
+import { SingleSelection } from "@react-types/shared"
+import classnames from "classnames"
+import { Label, FieldMessage } from "@kaizen/draft-form"
+import {
+  MenuTriggerProvider,
+  MenuTriggerProviderProps,
+} from "../../provider/MenuTriggerProvider"
+import {
+  SelectionProvider,
+} from "../../provider/SelectionProvider"
+import { ItemType } from "../../types"
+import { FloatingSelectWrapper } from "../FloatingSelectWrapper"
+import { HiddenSelectWrapper } from "../HiddenSelectWrapper/HiddenSelectWrapper"
+import rootStyles from "./rootStyles.module.scss"
+export interface RootProps extends MenuTriggerProps, SelectionProps {
+  trigger: React.ReactNode
+  children: React.ReactNode // the content of the menu
+  description?: React.ReactNode
+}
+
+type MenuTriggerProps = Omit<MenuTriggerProviderProps, "children">
+
+interface SelectionProps {
+  id: string // provide A11y context for label
+  label: string // provide A11y context for listbox
+  items: ItemType[]
+  selectedKey?: SingleSelection["selectedKey"]
+  onSelectionChange?: SingleSelection["onSelectionChange"]
+}
+
+export type SelectProps = RootProps
+
+export const Root: React.VFC<RootProps> = ({
+  id,
+  trigger,
+  children,
+  isOpen,
+  defaultOpen,
+  onOpenChange,
+  label,
+  items,
+  selectedKey,
+  onSelectionChange,
+  isDisabled,
+  isFullWidth,
+  description,
+}) => {
+  const menuTriggerProps = {
+    isOpen,
+    defaultOpen,
+    onOpenChange,
+    isDisabled,
+    isFullWidth,
+    description,
+    items,
+  }
+
+  const selectionProps = {
+    id,
+    label,
+    items,
+    selectedKey,
+    onSelectionChange,
+  }
+
+  return (
+    <div>
+      <Label
+        htmlFor={id}
+        aria-label={label}
+        classNameOverride={classnames([
+          !isFullWidth && rootStyles.notFullWidth,
+        ])}
+      >
+        {label}
+      </Label>
+      <MenuTriggerProvider {...menuTriggerProps}>
```

Comment:
Why does the MenuTriggerProvider wrap the entire select?

Reply:
aside from the div right underneath, all the other components use this context

Reply:
Ah okay I see now. I think we need to re-name this provider to something like `SelectProvider` since it's actually storing most of the data about the entire select, not just the menu trigger.

And by extension the `SelectionProvider` is a bit confusing as well, why not have it all in one provider?

Reply:
This provider only has context around the when the menu is open or not, SelectionProvider store the state of the menu items and what is item is selected, It could be possible to remove the contexts and use render props to pass the relevant informations but that would require some refactoring

Reply:
I think the context is okay to keep, since multiple parts of the internal components need to more or less be sync'd with the props provided, but I don't see the need for two of them.

Reply:
@gyfchong We can refactor this to be using one Select Provider but involves quite a bit of refactor. We were thinking this could be a next iteration card and then merge this in? Unless you think this work should be done on this PR

cc: @MrBaptastic